src/Symfony/Component/Form/Extension/Core/DataTransformer/BooleanToStringTransformer.php

```diff
@@ -80,6 +94,23 @@ public function reverseTransform($value)
             throw new TransformationFailedException('Expected a string.');
         }
 
-        return true;
+        if (in_array($value, array_merge($this->defaultTrueValues, array($this->trueValue)), true)) {
```

Comment:
Instead of the expensive expensive `array_merge()`, could you rearrange this to:

~~~php
if ($this->trueValue === $value || in_array($value, $this->defaultValues, true)) {
~~~